automatic/python3/update.ps1

```diff
@@ -12,52 +14,146 @@ function global:au_SearchReplace {
+      if (!$line) {
+          $copyright += "`n"
+          $reading_copyright = $false
+          continue
+      }
+      if ($line.StartsWith('Copyright')) {
+          if ($reading_copyright) {
+              $copyright += "`n"
+          }
+          $copyright += $line
+          $reading_copyright = $true
+      } elseif ($reading_copyright) {
+          $copyright += " $line"
+      } else {
+          break
+      }
+  }
+  $Latest.Copyright = $copyright.Trim()
+}
+
+function global:au_AfterUpdate($Package) {
+    $cdata = $Package.NuspecXml.CreateCDataSection($Latest.Copyright)
+    $xml_Copyright = $Package.NuspecXml.GetElementsByTagName('copyright')[0]
+    $xml_Copyright.RemoveAll()
+    $xml_Copyright.AppendChild($cdata) | Out-Null
+    $Package.NuspecXml.package.metadata.licenseUrl = $Latest.LicenseUrl
+    $Package.NuspecXml.package.metadata.title = $Latest.Title
+    $Package.SaveNuspec()
+    $Latest.License | Out-File './legal/LICENSE.txt'
+}
+
+function GetStreams() {
+  param($release_files)
 
-    $url32 = $download_page.links | ? { $_.href -match "python-3.+.(exe|msi)$" -and $_.href -notmatch "amd64" } | select -first 1 -expand href
-    $url64 = $download_page.links | ? href -match "python-3.+amd64\.(exe|msi)$" | select -first 1 -expand href
-    if (!$url32 -or !$url64) {
-      Write-Host "Skipping due to missing installer: '$version'"; return
+  $version_re = '3\.(?<minor>\d+)\.(?<micro>\d+)(?:(?<releaselevel>a|b|rc)(?<serial>\d+))?'
+  $re = '^python-(?<version>' + $version_re + ')(?<amd64>-amd64)?\.exe$'
+  # collect URL pairs for all Python versions
+  $all_versions = @{ }
+  foreach ($file in $release_files) {
+      $file_name = $file.url.Split('/')[-1]
+      if ($file_name -match $re) {
+          $version = $matches['version']
+          $amd64 = $matches['amd64']
+          if (!$all_versions.containsKey($version)) {
+              $all_versions[$version] = @{ }
+          }
+          if ($amd64) {
+              $all_versions[$version]['64'] = $file.url
+          } else {
+              $all_versions[$version]['86'] = $file.url
+          }
+      }
+  }
+
+  # find latest version of each Python minor (3.x) version
+  $latest_versions = @{ }
+  foreach ($version_data in $all_versions.GetEnumerator()) {
+    # skip release files that don't have both x64 and x86 installers
+    if ($version_data.Value.Count -ne 2) {
+      continue
+    }
+    $version = Get-Version $version_data.Name
+    if ($latest_versions.ContainsKey($version.Version.Minor)) {
+      $known_latest_version = Get-Version $latest_versions[$version.Version.Minor]
+    } else {
+      $known_latest_version = Get-Version '0.0.0'
+    }
+    if ($version -gt $known_latest_version) {
+      $latest_versions[$version.Version.Minor] = $version_data.Name
     }
+  }
+
+  # generate two version streams per version, one for python3 and one for python3x package
+  $streams = @{ }
+  foreach ($version_data in $latest_versions.GetEnumerator()) {
```

Comment:
We rarely use `foreach` loops in this repository and prefer piped calls.

If possible, change the foreach loop to `$latest_versions | ForEach-Object { }` instead.